DataScience/Analytics/prototype/landRegistry.py

```diff
@@ -0,0 +1,846 @@
+    # parsing gets really confused if region or county is in the line
+    counties = ('WEST MIDLANDS', 'WEST YORKSHIRE', 'S YORKSHIRE', 'N YORKSHIRE', 'W YORKSHIRE', 'W SUSSEX',
+                'E SUSSEX', 'KENT', 'SOUTH GLAMORGAN', 'MID GLAMORGAN', 'WEST GLAMORGAN', ' ESSEX', 'SURREY', 'SUFFOLK',
+                'CHESHIRE', 'CARMARTHENSHIRE', 'DERBYSHIRE', 'BERKSHIRE', 'YORKSHIRE', 'HEREFORDSHIRE', 'LINCOLNSHIRE',
+                'NOTTINGHAMSHIRE', 'OXFORDSHIRE', 'BUCKINGHAMSHIRE', 'SHROPSHIRE', 'DORSET', 'DEVON', 'SOMERSET',
+                'CORNWALL', 'CLEVELAND', 'NORFOLK', 'STAFFORDSHIRE', 'MIDDLESEX', 'MERSEYSIDE', 'NORTH HUMBERSIDE',
+                'SOUTH HUMBERSIDE', 'ISLE OF WIGHT', 'CUMBRIA', 'FLINTSHIRE', 'GLOUCESTERSHIRE', 'WILTSHIRE',
+                'DENBIGHSHIRE', 'TYNE AND WEAR', 'NORTHUMBERLAND', 'NORTHAMPTONSHIRE', 'WARWICKSHIRE', 'HAMPSHIRE',
+                'GWENT', 'NORFOLK', 'CHESHIRE', 'POWYS', 'LEICESTERSHIRE', 'NORTHAMPTONSHIRE', 'NORTHANTS',
+                'WORCESTERSHIRE', 'HERTFORDSHIRE', 'CAMBRIDGESHIRE', 'BEDFORDSHIRE', 'LANCASHIRE')
+
+    # use this for the counties so that e.g. ESSEX ROAD does not become just ROAD...
+    # todo: the regex is getting ridiculous, maybe do other way around i.e. country must be followed by postcode or
+    #       be the last component.
+    addRegex = '(?:\s)(?!ROAD|LANE|STREET|CLOSE|DRIVE|AVENUE|SQUARE|COURT|PARK|CRESCENT|WAY|WALK|HEOL|FFORDD|HILL|GARDENS|GATE|GROVE|HOUSE|VIEW|BUILDING|VILLAS|LODGE|PLACE|ROW|WHARF|RISE|TERRACE|CROSS|ENTERPRISE|HATCH)'
+
+    # remove county from address but add a column for it
+    df['County'] = None
+    for county in counties:
+        msk = df['ADDRESS2'].str.contains(county + addRegex, regex=True, na=False)
+        df.loc[msk, 'County'] = county
+        df['ADDRESS2'] = df['ADDRESS2'].str.replace(county + addRegex, '', case=False)
+
+    return df
+
+
+def parseInputData(df, expandSynonyms=True):
+    """
+    Parses the address information from the input data.
+
+    :param df: pandas dataframe containing ADDRESS column that is being parsed
+    :type df: pandas.DataFrame
+    :param expandSynonyms: whether to expand common synonyms or not
+    :type expandSynonyms: bool
+
+    :return: pandas dataframe where the parsed information has been included
+    :rtype: pandas.DataFrame
+    """
+    # normalise data so that the parser has the best possible chance of getting things right
+    df = _normalizeData(df, expandSynonyms=expandSynonyms)
+
+    # get addresses and store separately as an vector
+    addresses = df['ADDRESS2'].values
+    print('Parsing', len(addresses), 'addresses...')
+
+    # temp data storage lists
+    organisation = []
+    department = []
+    subbuilding = []
+    buildingname = []
+    buildingnumber = []
+    buildingsuffix = []
+    street = []
+    locality = []
+    town = []
+    postcode = []
+
+    # loop over addresses - quite inefficient, should avoid a loop
+    for address in tqdm(addresses):
```

Comment:
Just a comment:
I wouldn't create such huge methods, I would instead create many small ones with a method name instead of a comment (like the one here)
Huge methods are harder to maintain and to unit test
They are also much harder to read